src/component/dataZoom/SliderZoomView.ts

```diff
@@ -387,26 +388,35 @@ class SliderZoomView extends DataZoomView {
             const areaPoints = [[size[0], 0], [0, 0]];
             const linePoints: number[][] = [];
             const step = thisShadowExtent[1] / (data.count() - 1);
-            let thisCoord = 0;
+            const normalizationConstant = size[0] / (thisDataExtent[1] - thisDataExtent[0]);
+            let thisCoord = -step;
 
             // Optimize for large data shadow
             const stride = Math.round(data.count() / size[0]);
             let lastIsEmpty: boolean;
-            data.each([otherDim], function (value: ParsedValue, index) {
-                if (stride > 0 && (index % stride)) {
+
+            data.each([info.thisDim, otherDim], function (thisValue: ParsedValue, otherValue: ParsedValue, index) {
+                if (stride > 0 && (index % stride) && info.thisAxis.type !== 'time') {
```

Comment:
Avoid doing `info.thisAxis.type !== 'time'` check in the data loop. It may affect the performance when having large amount of data. Instead we can cache the check result outside the loop